### PR TITLE
Fix cart total display when no discounts are applied

### LINKED ISSUE #471: Cart total does not display properly when there's no discount applied
### Steps to Reproduce

On RC:
- Add something to the cart.
- Observe total - it will be a plain 0 (no currency formatting, and won't reflect the actual price of the cart)

### Expected Behavior

Cart total should be properly displayed with currency formatting.

### Actual Behavior

It doesn't. The display is correct when there's a discount applied. 

### Related Issues

#431

### Screenshot or Screencast

image

### PR DESCRIPTION
**Pull request by jkachel**
#### Pre-Flight checklist

- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?
#471 

#### What's this PR do?
Fixes #471 

#### How should this be manually tested?
Add an item to the cart - the total display should be correct. Alternatively, add an item to the cart and hit the cart API (`api/v0/checkout/cart/`) - the discounted_price should match the total_price.

### CODE REVIEW

File: ecommerce/serializers.py [hunk 1]
```diff
@@ -172,7 +172,9 @@ def get_discounted_price(self, instance):
         discounts = instance.discounts.all()
 
         if discounts.count() == 0:
-            return None
+            return sum(
+                basket_item.product.price for basket_item in instance.basket_items.all()
```
Comment (annagav) [inline ecommerce/serializers.py:176]: The only thing that concerns me here, is that in the discounted amount the the discounted price is being multiplied by the quantity attribute of BasketItem, and here it is just the price independently of quantity of items. Maybe let's make it consistent.

**Review by annagav — CHANGES_REQUESTED** [review]

File: ecommerce/serializers.py (hunk 1 above)
Comment (jkachel) [reply ecommerce/serializers.py]: Good call; I added a method to the BasketItems model to handle the base price (before discounts) of the item in https://github.com/mitodl/mitxonline/pull/472/commits/a5d67a2057a9ef96698896b90d553f53dc2170f8

File: ecommerce/serializers.py [hunk 2]
```diff
@@ -172,10 +172,14 @@ def get_discounted_price(self, instance):
...
         if discounts.count() == 0:
-            return None
+            return sum(
+                (basket_item.quantity * basket_item.product.price)
+                for basket_item in instance.basket_items.all()
+            )
 
         return sum(
-            basket_item.discounted_price for basket_item in instance.basket_items.all()
+            (basket_item.quantity * basket_item.discounted_price)
```
Comment (annagav) [inline ecommerce/serializers.py:181]: I think discounted_price is already price*quantity. I am looking here: https://github.com/mitodl/mitxonline/blob/main/ecommerce/models.py#L83

**Review by annagav — CHANGES_REQUESTED** [review]

File: ecommerce/serializers.py (hunk 2 above)
Comment (jkachel) [reply ecommerce/serializers.py]: Think I missed something in the last commit.. should be fixed now though.

**Pull request merged**